tests/test_metadata.py

```diff
@@ -0,0 +1,83 @@
+from mapreader import loader
+import pytest
+import os
+import pandas as pd
+import shutil
+
+@pytest.fixture
+def dir_name():
+    return "metadata_test_dir"
+
+#if there is extra info in the metadata
+def test_extra_metadata(dir_name):  
+    os.mkdir(dir_name)
+    files = ["file1.png", "file2.png"]
+    for file in files:
+        with open(f"{dir_name}/{file}", "w") as f:
+            pass
+    metadata_df = pd.DataFrame({"name":["file1.png", "file2.png", "file3.png"], "coord":[(1.1,1.5),(2.1,1.0),(3.1,4.5)], "other":["one","two","three"]})
+    metadata_df.to_csv(f"{dir_name}/metadata_df.csv", sep="|")
+    
+    my_files=loader(f"{dir_name}/*png")
+    assert len(my_files)==2
+    my_files.add_metadata(f"{dir_name}/metadata_df.csv")
+    assert my_files.images["parent"]["file2.png"]["other"]=="two"
+    assert isinstance(my_files.images["parent"]["file2.png"]["coord"], tuple)
+
+#if metadata info matches 
+def test_matching_metdata(dir_name):
+    with open(f"{dir_name}/file3.png", "w") as f:
+        pass
```

Comment:
I think that this assumes that these tests will execute in a particular order, which is something that should be avoided.

In this case `test_extra_metadata` needs to run before `test_matching_metdata` (if not f"{dir_name}/metadata_df.csv" will not exist).

Creating metadata_df.csv in a fixture would avoid the need to execute the tests in a specific order. Defining the scope of the fixtures can mean that it will only get executed once and the result(s) cached. See https://docs.pytest.org/en/latest/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session.

Reply:
think this is fixed now by using tmp_path.